src/ByteSequence.php

```diff
@@ -17,27 +17,27 @@
 interface ByteSequence
 {
     /**
-     *  UTF-8 BOM sequence.
+     * UTF-32 BE BOM sequence.
```

Comment:
the re-ordering is not the issue, the issue is in the implementation of `bom_match` the function should be updated by sorting the constant using their value length.

Reply:
I too thought relying on a definition order in the interface was flaky but it was the path of least resistance. And, the test would guard it from becoming out of order in the future.